### PR TITLE
add more comprehensive tests for incr. comp.

### PR DESCRIPTION
**Pull request by nikomatsakis**
Test various changes one can do to structs.

r? @michaelwoerister

### CODE REVIEW

File: src/test/incremental/struct_add_field.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+pub fn use_X(x: X) -> u32 {
+    x.x as u32
+}
+
+#[rustc_dirty(label="TypeckItemBody", cfg="rpass2")]
+pub fn use_EmbedX(embed: EmbedX) -> u32 {
+    embed.x.x as u32
+}
+
+#[rustc_dirty(label="TypeckItemBody", cfg="rpass2")] // FIXME(#33850) should be clean
```
Comment (michaelwoerister) [inline src/test/incremental/struct_add_field.rs]: Interesting...

**Comment by michaelwoerister** [comment]
@bors r+

Do we have some documentation somewhere that describes the dependency graph node labels? I.e. how do we expect people to know that it's specifically `TypeckItemBody` (for example) that should be invalidated by a given change?

**Comment by nikomatsakis** [comment]
There is no documentation aside from the code. But obviously these are unit
tests, so familiar with the code seems like a reasonable expectation to me.

On Tue, May 24, 2016 at 6:02 PM, Michael Woerister <notifications@github.com

**Comment by Manishearth** [comment]
Travis fails

**Comment by nikomatsakis** [comment]
closing in favor of a new PR that incorporates a fix.

**Pull request closed without merging**